ribbon-eureka/src/main/java/com/netflix/niws/loadbalancer/CompositeEurekaEnabledNIWSServerList.java

```diff
@@ -0,0 +1,42 @@
+package com.netflix.niws.loadbalancer;
+
+import java.util.List;
+
+import com.netflix.client.IClientConfigAware;
+import com.netflix.client.config.IClientConfig;
+import com.netflix.loadbalancer.ServerList;
+
+/**
+ * @author Tomasz Bak
+ */
+public class CompositeEurekaEnabledNIWSServerList implements ServerList<DiscoveryEnabledServer>, IClientConfigAware {
+
+    private DiscoveryEnabledNIWSServerList eureka1ServerList;
+    private Eureka2EnabledNIWSServerList eureka2ServerList;
+
+    @Override
+    public void initWithNiwsConfig(IClientConfig clientConfig) {
+        this.eureka1ServerList = new DiscoveryEnabledNIWSServerList();
+        this.eureka1ServerList.initWithNiwsConfig(clientConfig);
+
+        this.eureka2ServerList = new Eureka2EnabledNIWSServerList();
+        this.eureka2ServerList.initWithNiwsConfig(clientConfig);
+    }
+
+    @Override
+    public List<DiscoveryEnabledServer> getInitialListOfServers() {
+        return loadListOfServers();
+    }
+
+    @Override
+    public List<DiscoveryEnabledServer> getUpdatedListOfServers() {
+        return loadListOfServers();
+    }
+
+    private List<DiscoveryEnabledServer> loadListOfServers() {
```

Comment:
There is no need for reboot. For each request a status flag is checked, and the request is dispatched accordingly.